packages/env-options/src/env-options.js

```diff
@@ -0,0 +1,93 @@
+// @ts-check
+
+// `@endo/env-options` needs to be imported quite early, and so should
+// avoid importing from ses or anything that depends on ses.
+
+// /////////////////////////////////////////////////////////////////////////////
+// Prelude of cheap good - enough imitations of things we'd use or
+// do differently if we could depend on ses
+
+const { freeze } = Object;
+const { apply } = Reflect;
+
+// Should be equivalent to the one in ses' commons.js even though it
+// uses the other technique.
+const uncurryThis =
+  fn =>
+  (receiver, ...args) =>
+    apply(fn, receiver, args);
+const arrayFlatMap = uncurryThis(Array.prototype.flatMap);
+const arrayPush = uncurryThis(Array.prototype.push);
+const arrayJoin = uncurryThis(Array.prototype.join);
+
+const q = JSON.stringify;
+
+const Fail = (template, ...args) => {
+  const msg = arrayFlatMap(args, (arg, i) => [template[i], arg]);
+  arrayPush(msg, template[args.length]);
+  throw Error(arrayJoin(msg, ''));
+};
```

Comment:
How much do we care about performance here? Assuming "_not much because this only applies for generating exceptions_", I think the code could be a bit more clear. Some options:

```js
const Fail = (literals, ...args) => {
  const msg = arrayReduce(literals, (head, str, i) => `${head}${args[i - 1]}${str}`);
  throw Error(msg);
};
```


```js
const Fail = (literals, ...args) => {
  const msgParts = arrayMap(literals, (str, i) =>
    i === 0 ? str : `${args[i - 1]}${str}`
  );
  throw Error(arrayJoin(msgParts, ""));
};
```

```js
const Fail = (literals, ...args) => {
  let msg = literals[0];
  for (let i = 0; i < args.length; i += 1) {
    msg = `${msg}${args[i]}${literals[i + 1]}`;
  }
  throw Error(msg);
};
```

Reply:
I believe the last of these options to be likely both the fastest and clearest.

Reply:
Done